src/main/java/org/sagebionetworks/bridge/hibernate/HibernateSurveyDao.java

```diff
@@ -0,0 +1,365 @@
+        ImmutableMap<String, Object> params = ImmutableMap.of(
+                "studyKey", studyIdentifier.getIdentifier(),
+                "guid", guid);
+        
+        String getQuery = SELECT_TEMPLATE + ((!includeDeleted) ? GET_ACTIVE : GET_ALL);
+        
+        List<HibernateSurvey> results = hibernateHelper.queryGet(
+                getQuery, params, OFFSET, 50, HibernateSurvey.class);
+        
+        return ImmutableList.copyOf(results);
+    }
+
+    @Override
+    public Survey getSurveyMostRecentVersion(StudyIdentifier studyIdentifier, String guid) {
+        ImmutableMap<String, Object> params = ImmutableMap.of(
+                "studyKey", studyIdentifier.getIdentifier(),
+                "guid", guid);
+        
+        String getQuery = SELECT_TEMPLATE + GET_ACTIVE;
+        
+        List<HibernateSurvey> results = hibernateHelper.queryGet(
+                getQuery, params, OFFSET, 50, HibernateSurvey.class);
+        if (!results.isEmpty()) {
+            return results.get(0);
+        }
+        return null;
+    }
+
+    @Override
+    public Survey getSurveyMostRecentlyPublishedVersion(StudyIdentifier studyIdentifier, String guid,
+            boolean includeElements) {
+        if (guid.toLowerCase().startsWith(IDENTIFIER_PREFIX)) {
+            ImmutableMap<String, Object> params = ImmutableMap.of(
+                    "studyKey", studyIdentifier.getIdentifier(),
+                    "identifier", guid.substring(IDENTIFIER_PREFIX.length()),
+                    "published", 1);
+            
+            String getQuery = SELECT_TEMPLATE + GET_ACTIVE_PUBLISH_WITH_IDENTIFIER;
+            List<HibernateSurvey> results = hibernateHelper.queryGet(
+                    getQuery, params, OFFSET, 50, HibernateSurvey.class);
+            if (!results.isEmpty()) {
+                ImmutableMap<String, Object> elementParams = ImmutableMap.of(
+                        "surveyGuid", results.get(0).getGuid(),
+                        "createdOn", results.get(0).getCreatedOn());
+                
+                String getElementQuery = SELECT_ELEMENT_TEMPLATE + GET_ELEMENTS;
+                
+                List<HibernateSurveyElement> elementResults = hibernateHelper.queryGet(
+                        getElementQuery, elementParams, OFFSET, 50, HibernateSurveyElement.class);
+//                results.get(0).setElements(elementResults);
+                return results.get(0);
+            }
+        } else {
+            ImmutableMap<String, Object> params = ImmutableMap.of(
+                    "studyKey", studyIdentifier.getIdentifier(),
+                    "guid", guid,
+                    "published", 1);
+            
+            String getQuery = SELECT_TEMPLATE + GET_ACTIVE_PUBLISH;
+            List<HibernateSurvey> results = hibernateHelper.queryGet(
+                    getQuery, params, OFFSET, 50, HibernateSurvey.class);
```

Comment:
This API only gets 1 survey, so there's no need to ask Hibernate for 50 of them.